src/js/stellar/api.factory.js

```diff
@@ -528,18 +510,26 @@ myApp.factory('StellarApi', ['$rootScope', 'StellarHistory', 'StellarOrderbook',
             offerId : offer_id
           });
           const te = new StellarSdk.TransactionBuilder(account).addOperation(op).build();
-          return AuthenticationFactory.sign(te);
-        }).then((te) => {
-          return _server.submitTransaction(te);
+          return this._submitTransaction(te, account);
         }).then((txResult) => {
           console.log(txResult);
-          callback(null, txResult.hash);
+          callback(null, txResult);
         }).catch((err) => {
           console.error('Cancel Offer Fail !', err);
           callback(err, null);
         });
       },
 
+      _submitTransaction(transaction, account) {
```

Comment:
Moved all occurrences of server.submitTransaction to this helper method where it determines whether it can use StellarGuard or not.